frontend/src/metabase/questions/components/QuestionMoveToast/QuestionMoveToast.tsx

```diff
@@ -15,31 +19,26 @@ interface QuestionMoveToastProps {
   question: Question;
 }
 
-const getMessage = (question: Question, collectionLink: JSX.Element) => {
-  const type = question.type();
-
-  if (type === "question") {
-    return jt`Question moved to ${collectionLink}`;
-  }
-
-  if (type === "model") {
-    return jt`Model moved to ${collectionLink}`;
-  }
-
-  if (type === "metric") {
-    return jt`Metric moved to ${collectionLink}`;
-  }
-
-  throw new Error(`Unknown question.type(): ${type}`);
-};
-
 function QuestionMoveToast({ collectionId, question }: QuestionMoveToastProps) {
   const id = coerceCollectionId(collectionId);
-  const collectionLink = <CollectionLink key="collection-link" id={id} />;
+  const type = question.type();
+  const { data: collection } = useGetCollectionQuery({ id });
+  const collectionLink = collection ? (
+    <CollectionLink key="collection-link" to={Urls.collection(collection)}>
+      {collection.name}
+    </CollectionLink>
+  ) : (
+    <Fragment key="fragment" />
+  );
+
   return (
     <ToastRoot>
       <StyledIcon name="collection" />
-      {getMessage(question, collectionLink)}
+      {match(type)
```

Comment:
nit: looks more like a switch/case for me

Reply:
2b63f7f395365fae8deb3783d8d36b111ed7860e